### ISSUE TITLE
Documentation refactoring

### ISSUE THREAD

**Issue by lhanusov** [keycloak/keycloak-benchmark#167]
### Description

Current documentation layout - Github README.md doesn't reflect actual development and the user is not able to follow how to properly setup the environment, dataset import and trigger the performance benchmark and also some parts are still missing.

Proposed changes - chapter layout:
* environment provisioning
** default -> minikube automation
* dataset import
** basic setup is a part of the minikube automation
** automated way to load larger datasets
* performance benchmark
** automated script to run it with particular parameters e.g. --server-url, --test, --client-secret, etc.

**Comment by kami619** [comment]
@lhanusov perfect timing - take a look at this PR and add anything additional which is missing from your suggestion as a review comment :) https://github.com/keycloak/keycloak-benchmark/pull/166 this modifies the existing documentation which is splintered and the new documentation is live at https://www.keycloak.org/keycloak-benchmark/

**Comment by ahus1** [comment]
@kami619 - even after the change I can see that there should be a more step-by-step guide chaining the different projects together in a walk through. Still, get that other PR merged first, then use this issue to add what's missing.

**Comment by kami619** [comment]
Fair point, it would be needing more betterment in terms of chaining things.

**Comment by xgp** [comment]
I don't know if this is the right place for this, but there is a fairly large disconnect between the [Preparing Keycloak for testing](https://www.keycloak.org/keycloak-benchmark/benchmark-guide/latest/preparing-keycloak) manual steps and the next [Running benchmarks from the CLI](https://www.keycloak.org/keycloak-benchmark/benchmark-guide/latest/run/running-benchmark-cli) step. Unless you read the `manage_gatling_client.sh` script (which isn't included in the distribution), you won't know how to set the values correctly so that the tests will function properly.

**Comment by kami619** [comment]
@xgp https://www.keycloak.org/keycloak-benchmark/benchmark-guide/latest/preparing-keycloak#manual-configuration, the steps specified in this section are what's automated in the manage_gatling_client.sh. Can you please elaborate a little bit more on the disconnect you are facing ? Would love to collect the feedback to make it better :)

**Comment by xgp** [comment]
@kami619 Thanks for looking at this. I'm referring to the default values. 

In the manual steps in the documentation, it leaves it up to you to choose the realm name, the client name, and the client secret. Then, the next step doesn't tell you where to use/set the values you chose. There should be documentation to tell you where to set those values in the benchmark config.

If you use the script, it picks `realm-0` realm name, `gatling` client name, and `setup-for-benchmark` client secret, and then the scenarios "just work".

**Comment by kami619** [comment]
fair point, we'll update the docs to reflect suggested/used default values. Thanks for bringing it up, anything else you might face as a hurdle in connecting the dots, do log it as a separate issue. Thanks again :)

**Comment by tkyjovsk** [comment]
@kami619 This issue is probably outdated now, isn't it?

**Comment by kami619** [comment]
> @kami619 Thanks for looking at this. I'm referring to the default values.
> 
> In the manual steps in the documentation, it leaves it up to you to choose the realm name, the client name, and the client secret. Then, the next step doesn't tell you where to use/set the values you chose. There should be documentation to tell you where to set those values in the benchmark config.
> 
> If you use the script, it picks `realm-0` realm name, `gatling` client name, and `setup-for-benchmark` client secret, and then the scenarios "just work".

I think the issue @xgp pointed out still exists. We need to atleast address that before closing out the broader ticket.
